tests/Benchmark_BatchGuesser.cpp

```diff
@@ -0,0 +1,269 @@
+                    std::vector<Field> &evec,
+                    std::vector<Complex> &eval,
+                    const unsigned int evBatchSize,
+                    const unsigned int si, const unsigned int sf)
+{
+    GridBase *g       = in[0].Grid();
+    double   lVol     = g->lSites();
+    double   siteSize = sizeof(typename Field::scalar_object);
+    double   lSizeGB  = lVol*siteSize/1024./1024./1024.;
+    double   nIt      = evBatchSize*(sf - si);
+    double   t        = 0.;
+    
+    t -= usecond();
+    for (unsigned int i = 0; i < evBatchSize; ++i)
+    for (unsigned int j = si; j < sf; ++j)
+    {
+        axpy(out[j], TensorRemove(innerProduct(evec[i], in[j]))/eval[i], evec[i], out[j]);
+    }
+    t += usecond();
+    // performance (STREAM convention): innerProduct 2 reads + axpy 2 reads 1 write = 5 transfers
+    std::cout << GridLogMessage << "projAccumulate: " << t << " us | " << 5.*nIt*lSizeGB << " GB | " << 5.*nIt*lSizeGB/t*1.0e6 << " GB/s" << std::endl;
+
+
+};
+
+template<typename Field>
+void ProjAccumRunner(std::vector<Field> &in, std::vector<Field> &out, unsigned int eb, unsigned int sb)
+{
+    GridBase *g       = in[0].Grid();
+
+    std::vector<Field> eigenVec(1,g);
+    std::vector<Complex> eigenval;
+    
+    std::vector<int> seeds({1,2,3,4});
+    GridSerialRNG            RNG;   RNG.SeedFixedIntegers(seeds);
+    std::vector<int> seeds5({5,6,7,8});
+    GridParallelRNG          RNG5(g);  RNG5.SeedFixedIntegers(seeds5);
+
+
+    GridStopWatch w1;
+    GridTime ProjAccum = w1.Elapsed() - w1.Elapsed();
+    
+    unsigned int evSize = 4*eb;
+
+    std::cout << GridLogMessage << "ProjAccumRunner start" << std::endl;
+    
+    for (int i = 0; i < evSize; i += eb) // need to chnge loop structure.
+    {
+
+        unsigned int evBlockSize = std::min(evSize - i, eb);
+
+        // make new ev and eval of size evBlockSize to simulate ev IO.
+        
+        std::cout << GridLogMessage << "New eigenvector picks" << std::endl;
+
+        eigenVec.resize(evBlockSize,g);
+        eigenval.resize(evBlockSize);
+
+        for (auto &e: eigenVec)
+        {
+            random(RNG5,e);
+        }
+
+        //for (unsigned int e = 0; e < eigenval.size(); e++)
+        for (auto &e: eigenval)
+        {
+            //eigenval[e] = 1.; // replace with random pick if grid rng supports not vector structures
+            random(RNG,e);
+            std::cout << GridLogDebug << "eigenval random pick" << e << std::endl;
+        }
+
+        std::cout << GridLogMessage << "evBlockSize: " << evBlockSize << std::endl;
+
+        for(unsigned int j = 0; j < in.size(); j += sb)
+        {
+            unsigned int srcBlockSize = std::min((int)in.size() - j, sb);
+
+            std::cout << GridLogMessage << "srcBlockSize: " << srcBlockSize << std::endl;
+
+            w1.Start();
+            projAccumulate<Field>(in, out, eigenVec, eigenval, evBlockSize, j, j + srcBlockSize);
```

Comment:
This should call the function in the guesser

Reply:
old version of the benchmark see Hadrons/utilities